client/http/client.go

```diff
@@ -165,7 +165,7 @@ func (ci *clientInner) requestWithRetry(
 		addr = ci.pdAddrs[idx]
 		err = ci.doRequest(ctx, addr, reqInfo, headerOpts...)
 		if err == nil {
-			break
+			return nil
```

Comment:
seem to not make a difference.

Reply:
You are right. I misunderstand the error here.